testbins/test-hal-pio/src/main.rs

```diff
@@ -0,0 +1,234 @@
+#![no_std]
+#![no_main]
+
+extern crate aerugo;
+extern crate calldwell;
+extern crate cortex_m;
+extern crate cortex_m_rt as rt;
+
+use aerugo::{
+    hal::drivers::{
+        embedded_hal::digital::{OutputPin, StatefulOutputPin},
+        pio::{
+            pin::{DriveMode, OutputMode, PinMode, PinState, PullResistor},
+            IoPortMetadata, Pin, Port, SynchronousPort, PIOC,
+        },
+        pmc::config::PeripheralId,
+    },
+    Aerugo, InitApi, SystemHardwareConfig,
+};
+use calldwell::with_rtt_out;
+use rt::entry;
+
+fn validate_pin_is_driven_high(pin: &Pin<OutputMode>) {
+    if !pin.is_set_high().unwrap() {
+        panic!("Pin is driven high, yet `is_set_high` returns `false`");
+    }
```

Comment:
Those could be replaced with
`assert!(cond, "message")`

Reply:
fix'd